sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1087,6 +1087,14 @@ object SQLConf {
       .checkValue(v => Set(1, 2).contains(v), "Valid versions are 1 and 2")
       .createWithDefault(2)
 
+  val STOP_RUNNING_DUPLICATE_STREAM = buildConf("spark.sql.streaming.stopExistingDuplicateStream")
+    .doc("Running two streams using the same checkpoint location concurrently is not supported. " +
+      "In the case where multiple streams are started on different SparkSessions, access to the " +
+      "older stream's SparkSession may not be possible, and the stream may have turned into a " +
+      "zombie stream. When this flag is true, we will stop the old stream to start the new one.")
+    .booleanConf
+    .createWithDefault(true)
```

Comment:
Can we have `false` by default to avoid the behavior changes?

Reply:
Great question. Here's my argument why we should change it:
 1. This change is going into Spark 3.0, a release where we can actually break existing behavior (unless it is critical behavior which people depend on)
 2. The existing behavior was that any new start of a stream would fail, because an existing stream was already running. This is programming error on the user's part.
 3. However, there are legitimate cases, where a user would like to restart a new instance of the stream (because they upgrade the code for instance), but they have no way of stopping the existing stream, because it turns into a zombie.

I would argue that 3 is more common than 2, and including 1, this is where we can change behavior and mention in release notes.

Reply:
Great question. Here's my argument why we should change it:
 1. This change is going into Spark 3.0, a release where we can actually break existing behavior (unless it is critical behavior which people depend on)
 2. The existing behavior was that any new start of a stream would fail, because an existing stream was already running. This is programming error on the user's part.
 3. However, there are legitimate cases, where a user would like to restart a new instance of the stream (because they upgrade the code for instance), but they have no way of stopping the existing stream, because it turns into a zombie.

I would argue that 3 is more common than 2, and including 1, this is where we can change behavior and mention in release notes.

Reply:
+1 for the release notes.

Reply:
+1 one the name.